packages/components/src/font-size-picker/test/index.tsx

```diff
@@ -1,290 +1,434 @@
-			const fontSizes = [
-				{
-					name: 'Small',
-					slug: 'small',
-					size: 12,
-				},
-			];
-			const setFontSize = jest.fn(
-				( nextSize ) => ( fontSize = nextSize )
-			);
+	describe( 'with > 5 font sizes', () => {
+		const fontSizes = [
+			{
+				slug: 'tiny',
+				name: 'Tiny',
+				size: '8px',
+			},
+			{
+				slug: 'small',
+				name: 'Small',
+				size: '12px',
+			},
+			{
+				slug: 'medium',
+				name: 'Medium',
+				size: '16px',
+			},
+			{
+				slug: 'large',
+				name: 'Large',
+				size: '22px',
+			},
+			{
+				slug: 'x-large',
+				name: 'Extra Large',
+				size: '30px',
+			},
+			{
+				slug: 'xx-large',
+				// no name
+				size: '40px',
+			},
+		];
 
+		it( 'displays a select control', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
+			} );
 			render(
 				<FontSizePicker
-					fontSizes={ fontSizes }
-					value={ fontSize }
-					onChange={ setFontSize }
 					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
 				/>
 			);
-
-			toggleCustomInput( true );
-			const unitSelect = getUnitSelect();
-			const unitLabel = getUnitLabel();
-			const input = screen.getByLabelText( 'Custom', {
-				selector: 'input',
-			} );
-
-			input.focus();
-			fireEvent.change( input, { target: { value: 16 } } );
-
-			expect( unitSelect ).toBeFalsy();
-			expect( unitLabel ).toBeTruthy();
-			expect( fontSize ).toBe( 16 );
+			await user.click(
+				screen.getByRole( 'button', { name: 'Font size' } )
+			);
+			const options = screen.getAllByRole( 'option' );
+			expect( options[ 0 ] ).toHaveAccessibleName( 'Default' );
+			expect( options[ 1 ] ).toHaveAccessibleName( 'Tiny 8' );
+			expect( options[ 2 ] ).toHaveAccessibleName( 'Small 12' );
+			expect( options[ 3 ] ).toHaveAccessibleName( 'Medium 16' );
+			expect( options[ 4 ] ).toHaveAccessibleName( 'Large 22' );
+			expect( options[ 5 ] ).toHaveAccessibleName( 'Extra Large 30' );
+			expect( options[ 6 ] ).toHaveAccessibleName( 'xx-large 40' );
+			expect( options[ 7 ] ).toHaveAccessibleName( 'Custom' );
 		} );
 
-		it( 'should use units when fontSizes size values have units', () => {
-			let fontSize;
-			const fontSizes = [
-				{
-					name: 'Small',
-					slug: 'small',
-					size: '12px',
-				},
-			];
-			const setFontSize = jest.fn(
-				( nextSize ) => ( fontSize = nextSize )
-			);
+		test.each( [
+			{ value: undefined, expectedHint: 'Size' },
+			{ value: '8px', expectedHint: 'Size(8px)' },
+			{ value: '3px', expectedHint: 'Size(Custom)' },
+		] )(
+			'displays $expectedHint as header hint when value is $value',
+			( { value, expectedHint } ) => {
+				render(
+					<FontSizePicker
+						__nextHasNoMarginBottom
+						fontSizes={ fontSizes }
+						value={ value }
+					/>
+				);
+				// TODO: There's no accessible way to select the label.
+				const label = document.querySelector(
+					'.components-base-control__label'
+				);
+				expect( label ).toHaveTextContent( expectedHint );
```

Comment:
The `expectedHint` in that case doesn't look right to me. 

<img width="928" alt="Screen Shot 2022-10-28 at 12 20 17 pm" src="https://user-images.githubusercontent.com/6458278/198437472-0ea3c1b6-3b89-4686-abba-a5e6344cd436.png">

Could it be that the tests is passing because [toHaveTextContent](https://github.com/testing-library/jest-dom#tohavetextcontent) does partial matches?

If we're only interested in testing whether the accessible label and text hint are set correctly, then another option might be to just pass a single fontSize for the label you want to test:

```js
		test.each( [
			{
				value: undefined,
				expectedHint: 'Size Default',
				fontSizePresets: [ fontSizes[ 0 ] ],
			},
			{
				value: '8px',
				expectedHint: 'Size Tiny(px)',
				fontSizePresets: [ fontSizes[ 0 ] ],
			},
			{
				value: '3px',
				expectedHint: 'Size (Custom)',
				fontSizePresets: [ fontSizes[ 0 ] ],
			},
		] )(
			'displays $expectedHint as header hint when value is $value',
			( { value, expectedHint, fontSizePresets } ) => {
				render(
					<FontSizePicker
						__nextHasNoMarginBottom
						fontSizes={ fontSizePresets }
						value={ value }
					/>
				);
				const label = screen.getByLabelText( expectedHint );
				expect( label ).toBeInTheDocument();
				expect( label ).toHaveTextContent(
					// Removes whitespaces to match text interspersed in HTML elements.
					expectedHint.replace( /\s*/g, '' )
				);
			}
		);
```

Reply:
By fixing your other comment I think I inadvertently fixed this, good catch 😀